core/src/main/java/org/apache/iceberg/rest/RESTSessionCatalog.java

```diff
@@ -109,8 +110,11 @@ public RESTSessionCatalog() {
   }
 
   @Override
-  public void initialize(String name, Map<String, String> props) {
-    Preconditions.checkArgument(props != null, "Invalid configuration: null");
+  public void initialize(String name, Map<String, String> unresolved) {
```

Comment:
Nit / non-blocking: What about `unresolvedProps` or maybe the standard `props` and then using `fullProps` or some combination?

I don't love my ideas, but `unresolved` by itself is somewhat confusing for me. It initially reads as somehow more special than just the `properties` map (with or without env variable resolution applied).

By the interface, it is `props` so I'm ok with it, but the name itself is a little misleading when reading it imo.